lessons/06-tool-calling/README.md

```diff
@@ -0,0 +1,426 @@
+
+**You**: "Time beetle, can you help me upgrade Amelia's device?"
+
+**Time beetle**: "Of course! To upgrade Amelia's device, let's add the following tools to it:
+
+- **A tool than can**: Calculate the distance between two points on a map.
+- **A tool than can**: Figure out the GPS position of where Amelia is currently located.
+- **A tool than can**: Call an external API to get the weather forecast for a given location."  
+
+Here are the functions, all you need to do is to register them and test them out:
+
+```javascript
+function calculateDistance(lat1, long1, lat2, long2) {
+    // Perform the task of calculating the distance between two points
+    // Return the distance between the points
+    return Math.sqrt((lat2 - lat1) ** 2 + (long2 - long1) ** 2);
+}
+
+function getGpsPosition() {
+    // Perform the task of getting the GPS position of the current location
+    // Return the GPS position
+    return { lat: 7.5, long: 134.5 };
+}
+
+function getWeatherForecast(lat, long) {
+    // Perform the task of getting the weather forecast for a given location
+    // Return the weather forecast
+    return "Sunny";
+}
+```
+
+**You**: "Time beetle, are you sure these functions are going to work, looks like they're just returning some random values?"
+
+**Time beetle**: "That's correct, I can do the rest internally. All you need to do is to register them and test them out, make sure the AI model can call them."
+
+> Task: Register the `calculateDistance`, `getGpsPosition`, and `getWeatherForecast` tools with the AI model. Test the tools by calling them from the AI model. Use the code supplied in the previous sections as a reference.
+
+## Solution
+
+[Solution](./solution/solution.js)
+
+## Knowledge check
+
+**Question:**  
+What is the purpose of registering a tool with the AI model?
+
+A. To allow the AI model to directly execute the tool without developer intervention.  
+B. To provide metadata about the tool so the AI model can suggest its usage.  
+C. To replace the need for defining functions in the code.
+
+**Question:**  
+What is the role of tool metadata in tool calling?
+
+A. It describes the tool's purpose, inputs, and outputs for the AI model.  
+B. It provides the AI model with the tool's implementation details.  
+C. It ensures the tool is executed automatically by the AI model.
+
+**Question:**  
+Why do tool calling?
+
+A. To enable the AI model to perform tasks beyond its built-in capabilities by leveraging external functions.  
+B. To replace the need for human intervention in AI model development.  
+C. To allow the AI model to execute tools without requiring metadata.
+
+[Solution quiz](./solution/solution-quiz.md)
+
+## Self-Study resources
+
+- Explains the [process of tool calling](https://learn.microsoft.com/en-us/semantic-kernel/concepts/ai-services/chat-completion/function-calling/?pivots=programming-language-csharp)
+- Tool calling in the [framework Langchain](https://js.langchain.com/docs/how_to/tool_calling/)
```

Comment:
This should be:

- Tool calling in the [LangChain.js framework](https://js.langchain.com/docs/how_to/tool_calling/)